pydepsi/network.py

```diff
@@ -0,0 +1,178 @@
+"""Module for creating networks from STM points."""
+
+import math
+
+import numpy as np
+from scipy.spatial import Delaunay
+
+
+def get_distance(s, t):
+    """Calculate the distance between two points.
+
+    Args:
+    ----
+        s: the source point.
+        t: the target point.
+
+    Returns:
+    -------
+        The distance between the two points.
+    """
+    return math.dist(s, t)
+
+
+def check_in_range(value, min_length, max_length):
+    """Check whether a value is inside a specified range.
+
+    Args:
+    ----
+        value: float, value to check.
+        min_length: float, minimum allowed value, inclusive.
+        max_length: float, maximum allowed value, inclusive.
+
+    Returns:
+    -------
+        boolean: true if and only if the value is inside [min_length, max_length].
+    """
+    return min_length <= value and (max_length is None or value <= max_length)
+
+
+def generate_arcs(
+    stm_points, method="delaunay", x="lon", y="lat", min_length=0.00001, max_length=None, max_links=12, num_groups=8
+):
+    """Generate a network from a list of STM points.
+
+    The network is undirected and without self-loops.
+
+    Args:
+    ----
+        stm_points: Xarray.Dataset, input Space-Time Matrix.
+        method: method to form the network; either "delaunay" or "redundant".
+        x: str, first coordinate used to describe a point.
+        y: str, second coordinate used to describe a point.
+        min_length: float, minimum length of any generated arc.
```

Comment:
Please remove the min-length argument. This should be handled before the function is called, e.g. by the sparsify function. For example, if there are 2 corner reflectors close together, we want them both in the network (as parsed via the sparsify function).